pkg/db/kv_migration_test.go

```diff
@@ -0,0 +1,75 @@
+package db_test
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/pkg/db"
+	"github.com/treeverse/lakefs/pkg/db/params"
+	"github.com/treeverse/lakefs/pkg/kv"
+	kvpg "github.com/treeverse/lakefs/pkg/kv/postgres"
+	"github.com/treeverse/lakefs/pkg/testutil"
+)
+
+func TestKVMigration(t *testing.T) {
+	ctx := context.Background()
+	dbParams := params.Database{Driver: "pgx", ConnectionString: databaseURI, KVEnabled: true, DropTables: true}
+	kvStore, err := kv.Open(ctx, kvpg.DriverName, dbParams.ConnectionString)
+	testutil.MustDo(t, "Open KV store", err)
+	tests := []struct {
+		name       string
+		migrations map[string]kvpg.MigrateFunc
+		err        error
+		entries    int
+	}{
+		{
+			name:       "basic",
+			migrations: map[string]kvpg.MigrateFunc{"basic": testutil.MigrateBasic},
+			err:        nil,
+			entries:    5,
+		},
+		{
+			name:       "parallel",
+			migrations: map[string]kvpg.MigrateFunc{"basic": testutil.MigrateBasic, "parallel": testutil.MigrateParallel},
+			err:        nil,
+			entries:    10,
+		},
+		{
+			name:       "empty",
+			migrations: map[string]kvpg.MigrateFunc{"empty": testutil.MigrateEmpty},
+			err:        kv.ErrInvalidFormat,
+		},
+		{
+			name:       "no_header",
+			migrations: map[string]kvpg.MigrateFunc{"no_header": testutil.MigrateNoHeader},
+			err:        kv.ErrInvalidFormat,
+		},
+		{
+			name:       "bad_entry",
+			migrations: map[string]kvpg.MigrateFunc{"bad_entry": testutil.MigrateBadEntry},
+			err:        kv.ErrInvalidFormat,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			kvpg.UnregisterAll()
+			for n, m := range tt.migrations {
+				kvpg.RegisterMigrate(n, m, nil)
+			}
+			err = db.MigrateUp(dbParams)
+			require.ErrorIs(t, err, tt.err)
+
+			if tt.err == nil {
+				data, err := kvStore.Get(ctx, []byte(kv.DBSchemaVersionKey))
```

Comment:
We can call the public func `GetDBSchemaVersion` and verify the version and that err is nil?

Reply:
Done